### PR TITLE
shane/mdf: request.id of response to lsp server perhaps was not a number

### LINKED ISSUE #54: type-mismatch: request.id maybe string not num
```vim
def SendResponse(lspserver: dict<any>, request: dict<any>, result: dict<any>, error: dict<any>)
  var resp: dict<any> = lspserver.createResponse(request.id)
```

### PR DESCRIPTION
**Pull request by Shane-XB-Qian**
closes #54

### CODE REVIEW

File: autoload/lsp/lspserver.vim [hunk 1]
```diff
@@ -282,7 +282,11 @@ enddef
 
 # send a response message to the server
 def SendResponse(lspserver: dict<any>, request: dict<any>, result: dict<any>, error: dict<any>)
-  var resp: dict<any> = lspserver.createResponse(request.id)
+  if trim(string(request.id), "'") =~ '[^[:digit:]]\+' || trim(string(request.id), "'") == ''
```
Comment (yegappan) [inline autoload/lsp/lspserver.vim:285]: Do we need a check a for type(request.id) == v:t_string here before the call to trim()?

**Comment by Shane-XB-Qian** [comment]
if it's not a number, after string() and trim(), I supposed there should some other chars (not number chars), so probably ok.
// the vim9 cannot compare number with other types directly, so had to cast first.

--
shane.xb.qian
________________________________

File: autoload/lsp/lspserver.vim (hunk 1 above)
Comment (Shane-XB-Qian) [reply autoload/lsp/lspserver.vim]: thx review, i am changing the way you suggested.

**Comment by Shane-XB-Qian** [comment]
pls review again, i changed the way to you suggested.

**Comment by castedo** [comment]
To wrap up here, a proper fix for requests from an LSP server with a string id has just been merged:
https://github.com/yegappan/lsp/pull/768

**Pull request merged**